packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin2.tsx

```diff
@@ -543,8 +543,8 @@ export const TooltipPlugin2 = ({
 
       const event = plot!.cursor.event;
 
-      // if not viaSync, re-dispatch real event
-      if (event != null) {
+      // if not viaSync, re-dispatch real event and prevent unattached MouseEvent from being dispatched
+      if (event != null && event.currentTarget != null) {
```

Comment:
@leeoniya I have no idea if this breaks some case. When the tooltip was rendering, the event appearing here was a MouseEvent, which was then dispatched and caused the crosshair disappearing.

Reply:
sounds good, i'll check today

Reply:
doesnt seem to affect the behavior this code was added to fix, so LGTM.

i appreciate that you found a clean solution here :trophy:. this plugin's code is not the easiest to follow, but it needs to be really high perf, so there's a lot of React-ism stuff being juggled.

Reply:
upon further testing, it looks like `currentTarget` is always null, because it's only available during the event bubbling phase, but here we're dispatching the event after the fact. the reason it still worked is because setCursor below it still worked _almost_ the same way, but not quite.

i pushed a different fix with a comment explaining what was going on.